Purchases/Purchasing/StoreKitAbstractions/StoreProductDiscount.swift

```diff
@@ -0,0 +1,149 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  StoreProductDiscount.swift
+//
+//  Created by Joshua Liebowitz on 7/2/21.
+//
+
+import Foundation
+import StoreKit
+
+/// TypeAlias to StoreKit 1's Discount type, called `SKProductDiscount`
+@available(iOS 12.2, macOS 10.14.4, tvOS 12.2, watchOS 6.2, *)
+public typealias SK1ProductDiscount = SKProductDiscount
+
+/// TypeAlias to StoreKit 2's Discount type, called `StoreKit.Product.SubscriptionOffer`
+@available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+public typealias SK2ProductDiscount = StoreKit.Product.SubscriptionOffer
+
+@objc(RCStoreProductDiscount)
+public final class StoreProductDiscount: NSObject, StoreProductDiscountType {
+
+    @objc(RCPaymentMode)
+    public enum PaymentMode: Int {
+
+        case none = -1
+        case payAsYouGo = 0
+        case payUpFront = 1
+        case freeTrial = 2
+
+    }
+
+    private let discount: StoreProductDiscountType
+
+    init(_ discount: StoreProductDiscountType) {
+        self.discount = discount
+
+        super.init()
+    }
+
+    @objc public var offerIdentifier: String? { self.discount.offerIdentifier }
+    @objc public var price: Decimal { self.discount.price }
+    @objc public var paymentMode: PaymentMode { self.discount.paymentMode }
+    @objc public var subscriptionPeriod: SubscriptionPeriod { self.discount.subscriptionPeriod }
+
+    /// Creates an instance from any `StoreProductDiscountType`.
+    /// If `discount` is already a wrapped `StoreProductDiscount` then this returns it instead.
+    static func from(discount: StoreProductDiscountType) -> StoreProductDiscount {
+        return discount as? StoreProductDiscount
+        ?? StoreProductDiscount(discount)
+    }
+
+    public override func isEqual(_ object: Any?) -> Bool {
```

Comment:
does this one need to be `@objc`?

Reply:
`override` forces it to be, but I can add it to make it explicit.

Reply:
Actually I think it's redundant 😇